kong/db/migrations/core/005_120_to_130.lua

```diff
@@ -1,3 +1,139 @@
+local cassandra = require "cassandra"
+local cjson = require "cjson.safe"
+
+local function escape_literal(val)
+  if val == ngx.null then
+    return "NULL"
+  end
+
+  local t_val = type(val)
+  if t_val == "number" then
+    return tostring(val)
+  elseif t_val == "string" then
+    return "'" .. tostring((val:gsub("'", "''"))) .. "'"
+  elseif t_val == "boolean" then
+    return val and "TRUE" or "FALSE"
+  end
+  error("don't know how to escape value: " .. tostring(val))
+end
+
+
+local function add_upstream_algorithm_to_postgress(connector)
+  assert(connector:connect_migrations())
+
+  local select_query = [[
+    SELECT * FROM upstreams;
+  ]]
+
+  for row, err in connector:iterate(select_query) do
+    local algorithm
+    if row.hash_on == "none" then
+      algorithm = escape_literal("round-robin")
+    else
+      algorithm = escape_literal("consistent")
+    end
```

Comment:
Migrations should be idempotent: if an `algorithm` field already has a value, it should not be overwritten (in that case the iteration can just `goto continue`).